### PR TITLE
Custom elements updates, first cut.

### PR DESCRIPTION
**Pull request by arthurevans**

### CODE REVIEW

File: app/3.0/docs/devguide/custom-elements.md [hunk 1]
```diff
@@ -241,22 +263,27 @@ The custom element state isn't exposed as a property, but you can style elements
...
+`:defined` pseudo-class selector to target elements that are defined. **In native shadow DOM**,
+you can use this to provide placeholder styles for elements before they're upgraded:
 
 ```
 my-element:not(:defined) {
   background-color: blue;
 }
 ```
 
+**`:defined` is not supported by the Custom Elements polyfill.** See the [documentation on styling](style-shadow-dom#style-undefined-elements) for a workaround.
```
Comment (arthurevans) [inline app/3.0/docs/devguide/custom-elements.md]: This link doesn't seem to exist anymore?

**Comment by arthurevans** [comment]
Staged: https://3-0-custom-elements-dot-polymer-project.appspot.com

File: app/3.0/docs/devguide/custom-elements.md (hunk 1 above)
Comment (katejeffreys) [reply app/3.0/docs/devguide/custom-elements.md]: I put it back in https://github.com/Polymer/docs/pull/2476/commits/d9d4ff6473c85393def6d755d7745b661f3d9d63

File: app/3.0/docs/devguide/custom-elements.md [hunk 2]
```diff
@@ -16,45 +12,37 @@ Custom elements provide a component model for the web. The custom elements speci
...
 on Web Fundamentals.
 
-To define a custom element, you create an ES6 class and associate it with the custom element name.
+To define a custom element, you create an ES6 class and associate it with the custom element name. For the full set of Polymer features, extend the `PolymerElement` class:
 
-```
-// Create a class that extends HTMLElement (directly or indirectly)
-class MyElement extends HTMLElement { … };
+```js
+import {PolymerElement} from '@polymer/polymer/polymer-element.js`;
```
Comment (katejeffreys) [inline app/3.0/docs/devguide/custom-elements.md]: backtick should be single quote

File: app/3.0/docs/devguide/registering-elements.md [hunk 3]
```diff
@@ -8,27 +8,31 @@ title: Define an element
...
 
 By specification, the custom element's name **must start with a lower-case ASCII letter and must
 contain a dash (-)**.
 
 Example: { .caption }
 
-```
+```js
+// Import PolymerElement class
+import {PolymerElement} from '@polymer/polymer/polymer-element.js`;
```
Comment (katejeffreys) [inline app/3.0/docs/devguide/registering-elements.md]: backtick should be single quote

**Review by ghost — APPROVED** [review]
I am a little confused by what a legacy element is - do we have a definition of that?

File: app/3.0/docs/devguide/custom-elements.md (hunk 2 above)
Comment (arthurevans) [reply app/3.0/docs/devguide/custom-elements.md]: Fixed, ty.

**Pull request merged**